src/middleware/serializer/users.serializer.ts

```diff
@@ -7,20 +7,20 @@ import { Unknown } from '../../utils/apiResponse.utils'
 
 export default function (req: Request, res: Response, next: NextFunction) {
   // No data applied at the controller level
-  if (!req.user && !req.users) return res.status(400).send(Unknown)
+  if (!req.user && !req.users) return res.status(400).json(Unknown)
 
   const response = req.user ? serialize(req.user) : req.users.map(serialize)
 
-  res.status(req.statusCode).send(response)
+  res.status(req.statusCode).json(response)
 }
 
 function serialize(user: User): UserType {
   return {
     id: user.id,
     schoolId: user.schoolId,
     email: user.email,
-    createdAt: user.createdAt.toISOString(),
-    updatedAt: user.updatedAt.toISOString(),
+    createdAt: user.createdAt ? user.createdAt.toISOString() : null,
```

Comment:
Would we have a case where a User exists to be serialized without valid Created/Updated fields that wouldn't be considered invalid?

Reply:
Usually no. But because we're using the meta fields created by typeorm for created/updated/deleted at, you can create an in memory object that doesn't have it's createdAt or updatedAt on them yet. In production, those fields in this context (user serializer) should always be populated.

The reason for this (I think) is because typeorm doesn't want to add that info until the point it's inserted into the DB (which makes sense), but is annoying in testing. 

That's why in the `users.serializer` tests we have to manually add a `new Date` onto the expected data, to allow it act more like DB records.

Reply:
Would we not want the serializer to return an error then if the user object passed to it doesnt have proper fields (test or otherwise?) instead of allowing null fields?

Reply:
:thinking: that's no unreasonable.....

Idk if I'd say this is wrong, but I do see where you're coming from. Sure, let's do it. 

We just won't be able to serialize an in memory only user (without having hit the db first). If the time comes where we want to do that, we can update this then.